src/usecases/periodToString.ts

```diff
@@ -0,0 +1,155 @@
+export const periodToString = (schedules: CourseSchedule[]): string => {
+  /** module(9) × day(6) × period(8)を表現 */
+  const regularSchedules: {
+    [key in Exclude<CourseModule, "Unknown">]: [
+      number,
+      number,
+      number,
+      number,
+      number,
+      number
+    ];
+  } = {
+    Annual: [0, 0, 0, 0, 0, 0],
+    SpringA: [0, 0, 0, 0, 0, 0],
+    SpringB: [0, 0, 0, 0, 0, 0],
+    SpringC: [0, 0, 0, 0, 0, 0],
+    SummerVacation: [0, 0, 0, 0, 0, 0],
+    FallA: [0, 0, 0, 0, 0, 0],
+    FallB: [0, 0, 0, 0, 0, 0],
+    FallC: [0, 0, 0, 0, 0, 0],
+    SpringVacation: [0, 0, 0, 0, 0, 0],
+  };
+
+  const specialSchedules: {
+    [key in Exclude<CourseModule, "Unknown">]?: SpecialDay;
+  } = {};
+  /**
+   * ex.) { SpringC: "応談j" }
+   */
+
+  schedules.forEach((schedule) => {
+    if (schedule.module === "Unknown" || schedule.day === "Unknown") return;
+    if (weeks.includes(schedule.day)) {
+      if (schedule.period < 1 || 8 < schedule.period) return;
+      regularSchedules[schedule.module][weekdayNum(schedule.day)] |=
+        1 << (schedule.period - 1);
+    } else {
+      specialSchedules[schedule.module] = specialDayMap[schedule.day];
+    }
+  });
+
+  const convertedRegularSchedules = courseModules.reduce(
+    (convertedRegularSchedules, module) => {
+      // moduleごとにdayAndPeriodを作成
+      convertedRegularSchedules[module] = regularSchedules[
+        module
+      ].reduce<string>(
+        (dayAndPeriod, periodFlg, index, self) => {
+          // dayをみる
+          if (periodFlg === 0) return dayAndPeriod;
+          let tmp: string = dayToJa(weeks[index]);
+          let idx = self.indexOf(periodFlg, index + 1);
+          while (idx !== -1) {
+            tmp += dayToJa(weeks[idx]);
+            self[idx] = 0;
+            idx = self.indexOf(periodFlg, idx + 1);
+          }
+          // periodをみる
+          let period: string = "";
+          for (let i = 0; i < 8; i++) {
+            if ((periodFlg & (1 << i)) === 0) continue;
+            const start = i;
+            while ((periodFlg & (0b111 << i)) === 0b111 << i) i++;
+            period += period ? "," : "";
+            period += start !== i ? `${start + 1}-${++i + 1}` : i + 1;
+          }
+          tmp += period;
+
+          dayAndPeriod += dayAndPeriod ? " " + tmp : tmp;
+          return dayAndPeriod;
+        },
+        "" // ex.) "月1,2" や "水1 木3"
+      );
+      return convertedRegularSchedules;
+    },
+    {
+      Annual: "",
+      SpringA: "",
+      SpringB: "",
+      SpringC: "",
+      SummerVacation: "",
+      FallA: "",
+      FallB: "",
+      FallC: "",
+      SpringVacation: "",
+    }
+    /**
+     * ex.)
+     * {
+     *   Annual: "",
+     *   SpringA: "月1,2",
+     *   SpringB: "月1,2",
+     *   SpringC: "水1 木3",
+     *   SummerVacation: "",
+     *   FallA: "",
+     *   FallB: "",
+     *   FallC: "",
+     *   SpringVacation: "",
+     * }
+     */
+  );
+
+  return courseModules
+    .reduce<[string, string][]>(
```

Comment:
配列系こっちでやったほうが可読性は上がりそうやね
```suggestion
    .reduce<{module: string, date: string}[]>(
```